### DISCUSSION TITLE
Why logback lost logger when use MessageFormatter "{}"?

### DISCUSSION THREAD

**Discussion by Lihuanghe** [qos-ch/logback#597] [Q&A]
JDK: java8
os:windows
logback version: 1.2.3

This test case lost this line text logger :  "`>>>>>>>>this will lost<<<<<<<<<`"

```
public class TestLogbackBug {
	private static final Logger logger = LoggerFactory.getLogger(TestLogbackBug.class);
	
	@Test
	public void test() {
		ClassA c = new ClassA();
		logger.debug("A1 =========== c: {}",c);
	}
	class ClassA {
		@Override
		public String toString() {
			
			logger.error(">>>>>>>>this will lost<<<<<<<<<");
			return "ClassA [toString()=]";
		}
	}
}

```

OutPut:

```
18:40:54.016 [main] DEBUG com.TestLogbackBug - A1 =========== c: ClassA [toString()=]

```

**Comment by ceki** [comment]
Which appender are you using?

**Reply by Lihuanghe** [reply]
ch.qos.logback.core.ConsoleAppender

**Reply by Lihuanghe** [reply]
`ch.qos.logback.core.rolling.RollingFileAppender`   lost also.

**Comment by ceki** [comment]
UnsynchronizedAppenderBase has a guard protecting against recursive entry. See 
[UnsynchronizedAppenderBase line 66](https://github.com/qos-ch/logback/blob/master/logback-core/src/main/java/ch/qos/logback/core/UnsynchronizedAppenderBase.java#L66)
